api/management/commands/cleanup_question_stats.py

```diff
@@ -0,0 +1,221 @@
+        question_feedbacks_df["created_date"] = [
+            d.date() for d in question_feedbacks_df["created"]
+        ]
+        question_feedbacks_df["created_hour"] = [
+            d.time().hour for d in question_feedbacks_df["created"]
+        ]
+        # get list of unique dates
+        date_list = question_feedbacks_df["created_date"].unique()
+        print(f"{len(date_list)} unique dates")
+
+        # loop on unique dates
+        for date_unique in date_list:
+            daily_stat, created = DailyStat.objects.get_or_create(date=date_unique)
+            date_df = question_feedbacks_df[
+                question_feedbacks_df["created_date"] == date_unique
+            ]
+            # get number of feedbacks
+            date_feedback_count = date_df.shape[0]
+            date_feedback_from_quiz_count = date_df[date_df["source"] == "quiz"].shape[
+                0
+            ]
+            # update daily stat
+            daily_stat.question_feedback_count += date_feedback_count
+            daily_stat.question_feedback_from_quiz_count += (
+                date_feedback_from_quiz_count
+            )
+
+            # get list of unique date hours
+            date_hour_list = date_df["created_hour"].unique()
+            # loop on unique hours
+            for date_hour_unique in date_hour_list:
+                date_hour_df = date_df[date_df["created_hour"] == date_hour_unique]
+                # get number of feedbacks
+                date_hour_feedback_count = date_hour_df.shape[0]
+                date_hour_feedback_from_quiz_count = date_hour_df[
+                    date_hour_df["source"] == "quiz"
+                ].shape[0]
+                # update daily stat
+                daily_stat.hour_split[str(date_hour_unique)][
+                    "question_feedback_count"
+                ] += date_hour_feedback_count
+                daily_stat.hour_split[str(date_hour_unique)][
+                    "question_feedback_from_quiz_count"
+                ] += date_hour_feedback_from_quiz_count
+
+            # save daily stat
+            daily_stat.save()
+
+        # delete all processed question_feedbacks
+        question_feedbacks.delete()
+        print("QuestionFeedback deleted")
+
+
+def cleanup_quiz_stats():
+    """
+    cleanup QuizStat
+
+    TODO: how to keep score ?
+    """
+    pass
+
+
+class Command(BaseCommand):
+    """
+    python manage.py cleanup_question_stats
+    """
+
+    help = """Group and clean question stats"""
+
+    def handle(self, *args, **kwargs):
+        cleanup_question_stats()
+        cleanup_question_feedbacks()
+        # cleanup_quiz_stats()
+
+
+"""
+Daily stats
+- total number of answers
+- total number of answers from questions
+- total number of answers from quizs
+- total number of quizs played
+- total number of feedbacks (like/dislike)
+answers per hour ?
+
+
+Why pandas ?
+to do queries in memory instead of querying the database each time
```

Comment:
yes 🙏 
and also because it's far more readable and concise than ORM code. Much closer to SQL in that sense.
you will thank yourself in a few months 👅